src/cclib/io/wfxwriter.py

```diff
@@ -0,0 +1,475 @@
+        to_bohr = lambda x: utils.convertor(x, 'Angstrom', 'bohr')
+        nuc_coords = [coord_template % tuple(to_bohr(coord))
+                      for coord in self.ccdata.atomcoords[-1]]
+        return nuc_coords
+
+    def _net_charge(self):
+        """Net charge on molecule."""
+        return '%22.11E' % self.ccdata.charge
+
+    def _no_electrons(self):
+        """Number of electrons in molecule."""
+        homos = self.ccdata.homos
+        charge = self.ccdata.charge if hasattr(self.ccdata, 'charge') else 0.0
+        if len(homos) > 1:
+            return int(sum(homos)) + 2
+        else:
+            return int((homos[0] + 1)*2 - (charge % 2))
+
+    def _no_alpha_electrons(self):
+        """Number of alpha electrons in molecule."""
+        return int(numpy.ceil(self._no_electrons() / 2.0))
+
+    def _no_beta_electrons(self):
+        """Number of beta electrons in molecule."""
+        return self._no_electrons() - self._no_alpha_electrons()
+
+    def _spin_mult(self):
+        """Electronic Spin Multiplicity"""
+        return self.ccdata.mult
+
+    def _prim_centers(self):
+        """List of nuclear numbers upon which the primitive basis functions
+        are centered."""
+        prim_centers = []
+        for nuc_num, atom in enumerate(self.ccdata.gbasis, start=1):
+            for prims in atom:
+                prim_centers += [nuc_num] * self.num_orb[prims[0]]\
+                                * len(prims[1])
+
+        return self._list_format(prim_centers, 10, '%d ')
+
+    def _rearrange_modata(self, data):
+        """Rearrange MO related data."""
+        prim_types = self._prim_types()
+        if isinstance(data, numpy.ndarray):
+            data = data.tolist()
+
+        pos_yyx = [key for key, val in enumerate(prim_types)
+                   if val == 17]
+        pos_yyz = [key for key, val in enumerate(prim_types)
+                   if val == 16]
+
+        if pos_yyx:
+            for pos in pos_yyx:
+                data.insert(pos-3, data.pop(pos))
+        if pos_yyz:
+            for pos in pos_yyz:
+                data.insert(pos+3, data.pop(pos + 1))
+
+        return data
+
+
+    def _prim_types(self):
+        """Primitive type numbering starts at 1.
+        The definition of the Cartesian Gaussian primitive types is:
+        1 S, 2 PX, 3 PY, 4 PZ, 5 DXX, 6 DYY, 7 DZZ, 8 DXY, 9 DXZ, 10 DYZ,
+        11 FXXX, 12 FYYY, 13 FZZZ, 14 FXXY, 15 FXXZ, 16 FYYZ, 17 FXYY,
+        18 FXZZ, 19 FYZZ, 20 FXYZ,
+        21 GXXXX, 22 GYYYY, 23 GZZZZ, 24 GXXXY, 25 GXXXZ, 26 GXYYY,
+        27 GYYYZ, 28 GXZZZ,
+        29 GYZZZ, 30 GXXYY, 31 GXXZZ, 32 GYYZZ, 33 GXXYZ, 34 GXYYZ,
+        35 GXYZZ,
+        36 HZZZZZ, 37 HYZZZZ, 38 HYYZZZ, 39 HYYYZZ,
+        40 HYYYYZ, 41 HYYYYY, 42 HXZZZZ, 43 HXYZZZ, 44 HXYYZZ,
+        45 HXYYYZ, 46 HXYYYY, 47 HXXZZZ, 48 HXXYZZ, 49 HXXYYZ,
+        50 HXXYYY, 51 HXXXZZ, 52 HXXXYZ, 53 HXXXYY, 54 HXXXXZ, 55 HXXXXY,
+        56 HXXXXX
+        """
```

Comment:
I could test that for `GAMESS` and there isn't any difference. And `.wfx` files written for other programs (which use cartesian basis) are same as those of `GAMESS` .

As of now writer isn't expected to work for programs which use spherical basis. However, the files which do get written have ordering differences in primitive types, and therefore different values of normalized MO coefficients. 
Though they all pass the strict consistency checks by AIMALL program, shouldn't be considered correct.